workflow-core/src/commonMain/kotlin/com/squareup/workflow1/NullableInitBox.kt

```diff
@@ -0,0 +1,28 @@
+package com.squareup.workflow1
+
+import kotlin.jvm.JvmInline
+
+/**
+ * Used to wrap immutable nullable values whose holder may not yet be initialized.
+ * Check [isInitialized] to see if the value has been assigned.
+ */
+@JvmInline
+public value class NullableInitBox<T>(private val _value: Any? = Uninitialized) {
```

Comment:
I didn't take in at first that this is a `value class`. Cool!

Reply:
That was a @zach-klippenstein special when we were noodling on this.